composer/trainer/trainer.py

```diff
@@ -625,106 +875,37 @@ def __init__(
         torch_prof_with_flops: bool = True,
         torch_prof_num_traces_to_keep: int = -1,
     ):
-        # surpressing GradScaler warnings as they are always created
-        # self._use_grad_scaling() will raise a RuntimeError if grad scaling is not available when it is required
-        warnings.filterwarnings(action="ignore", message="torch.cuda.amp.GradScaler")
-
-        # ScaleSchedule is a deprecated algorithm, but if it is used, updated SSR with its ratio.
-        # TODO(#434): Remove this completely.
-        for algorithm in ensure_tuple(algorithms):
-            if isinstance(algorithm, ScaleSchedule):
-                scale_schedule_ratio = algorithm.ratio
-
-        if isinstance(max_duration, str):
-            max_duration = Time.from_timestring(max_duration)
-        elif isinstance(max_duration, int):
-            max_duration = Time.from_epoch(max_duration)
-
-        orig_max_duration = max_duration
-
-        if scale_schedule_ratio != 1.0:
-            max_duration = cast(Time[int], orig_max_duration * scale_schedule_ratio)
-            log.info(f'max_duration changed from {orig_max_duration} to {max_duration}')
-            if max_duration.value == 0:
-                raise ValueError(
-                    'Scale schedule has reduced the max_duration to 0. Set a higher ratio or use more epochs.')
-
-        if isinstance(deepspeed_config, bool):
-            self._deepspeed_config = {} if deepspeed_config else None
-        else:
-            self._deepspeed_config = deepspeed_config
-
-        if not device:
-            self._device = DeviceCPU() if not self.deepspeed_enabled else DeviceGPU()
-        elif isinstance(device, str):
-            if device == 'cpu':
-                self._device = DeviceCPU()
-            elif device == 'gpu':
-                self._device = DeviceGPU()
-            else:
-                raise ValueError(f'device ({device}) must be one of (cpu, gpu).')
-        else:
-            if not isinstance(device, Device):
-                raise ValueError('device must be of class Device')
-            self._device = device
-
-        if self.deepspeed_enabled or dist.get_world_size() > 1:
+        # Determine whether DeepSpeed is enabled
+        if deepspeed_config is True:
```

Comment:
This won't work, as `deepspeed_config` could also be a dictionary with parameters already specified. The one-liner would have to be something like `deepspeed_config = {} if deepspeed_config is True else None if deepspeed_config is False else deepseed_config` which I would have trouble understanding :D (the explicit True/False checks are required since, if the user passes in an empty dictionary, that should enable deepspeed, but `bool({}) is False`)

Reply:
Sounds good; we can simplify like that!

Reply:
Done